core/src/main/java/org/apache/druid/data/input/impl/DimensionsSpec.java

```diff
@@ -69,11 +71,22 @@ public static DimensionSchema convertSpatialSchema(SpatialDimensionSchema spatia
     return new NewSpatialDimensionSchema(spatialSchema.getDimName(), spatialSchema.getDims());
   }
 
+  public static Builder builder()
+  {
+    return new Builder();
+  }
+
+  public DimensionsSpec(List<DimensionSchema> dimensions)
+  {
+    this(dimensions, null, null, false);
+  }
+
   @JsonCreator
-  public DimensionsSpec(
+  private DimensionsSpec(
       @JsonProperty("dimensions") List<DimensionSchema> dimensions,
       @JsonProperty("dimensionExclusions") List<String> dimensionExclusions,
-      @Deprecated @JsonProperty("spatialDimensions") List<SpatialDimensionSchema> spatialDimensions
+      @Deprecated @JsonProperty("spatialDimensions") List<SpatialDimensionSchema> spatialDimensions,
+      @JsonProperty("includeAllDimensions") boolean includeAllDimensions
```

Comment:
Nit: I forget the jackson behaviour, but does having a primitive `boolean` here break the deserialization if the flag `includeAllDimensions` is absent?

Reply:
`boolean` variables are initialized to false when they are missing.